backend/python/app/services/implementations/file_service.py

```diff
@@ -0,0 +1,42 @@
+import os
+
+from ...models.file import File
+from ...models import db
+from ..interfaces.file_service import IFileService
+
+
+class FileService(IFileService):
+    def __init__(self, logger):
+        self.logger = logger
+
+    def get_file(self, id):
+        file = File.query.get(id)
+        if file is None:
+            self.logger.error(f"Invalid id {id}")
+            raise Exception(f"Invalid id {id}")
+        return file.to_dict()
+
+    def generate_file_name(self, filename, suffix=0):
+        upload_folder_path = os.getenv("UPLOAD_PATH")
+        if os.path.isfile(os.path.join(upload_folder_path, filename)):
+            split_filename = filename.split(".", 1)
+            name = f"{split_filename[0]}_{suffix}"
+            new_filename = ".".join([name, split_filename[1]])
+        if not os.path.isfile(os.path.join(upload_folder_path, new_filename)):
+            return os.path.join(upload_folder_path, new_filename)
+        else:
+            suffix += 1
+            return self.generate_file_name(filename, suffix)
```

Comment:
Can we use a loop for trying different suffixes, rather than recurse? We should avoid using up stack space where uneeded